### DISCUSSION TITLE
Rendering to/as 3D?

### DISCUSSION THREAD

**Discussion by soup587** [Techokami/SonicWorldsNext#103] [Q&A]
pretty new to godot, i've spent a week or so experimenting with sonic worlds and i'm wondering how i'd render a 3d scene with a player in it? i've noticed mentions of camera3d in the player script but it seems like it's almost entirely camera2d related code

**Comment by Renhoex** [comment]
That's weird, it might have been a comment push I didn't notice or something I added when I was sleep deprived and wasn't paying attention to my code comments.

but as for the 3D, there's none in this framework, it is possible to implement 3D stuff into the game because godot does support that but that's a whole can of worms that's you're probably going to have to research and if you're after a 3D sonic template I'm sorry to say but this isn't the framework if you're looking for a 3D sonic project.

**Reply by soup587** [reply]
i was thinking either something like superstars or just rendering the 2d sprites and physics in a 3d environment, i've made an attempt at at least 'copying' over the positions of various things like the player, environment and entities into a 3d space but the player script seems to override a 3d camera so it just renders in 2d anyway

**Reply by DimensionWarped** [reply]
I get what you want to do, but it's just outside the scope of this project. I guess you *might* be able to swap the 2D colliders on everything for 3D colliders, position all your terrain so that it has collision at z=0 and just hope it magically works (at least until you want to do something interesting with the playfield), but I have no idea if that's actually realistic at all.

**Reply by Renhoex** [reply]
If it's still showing the 2D camera it's because the player spawns a localized camera to grab focus,
It's in the player script around line 258
`camera.enabled = (playerControl == 1)`
you'll want to change  that to
`camera.enabled = false`
or remove it entirely.
